src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorProjectItem.cs

```diff
@@ -40,6 +42,8 @@ public DefaultRazorProjectItem(string basePath, string filePath, string relative
 
         public override string FileKind => _fileKind ?? base.FileKind;
 
+        public override string CssScope { get; }
```

Comment:
Why does the css scope need to be on the project item?

Reply:
It's a per-document value. `GenerateCommand` knows the value from its MSBuild inputs, but to get this information into the `CodeDocument` it has to go via the `ProjectItem`, because that's how all the per-document values get into the compiler.

If I'm missing a different way this could/should be done, please let me know!

Reply:
Spent some time to look it over more in depth and there's not currently an easy way to change it without doing a more structural change to how everything at the rzc layer works. Makes me sad but what you have is the best fit for the current architecture.